include/circt/Dialect/RTG/IR/RTGOps.td

```diff
@@ -455,6 +455,45 @@ def BagUniqueSizeOp : RTGOp<"bag_unique_size", [Pure]> {
   }];
 }
 
+//===- Array Operations -------------------------------------------------===//
+
+def ArrayCreateOp : RTGOp<"array_create", [
+  Pure,
+  SameTypeOperands,
+]> {
+  let summary = "create an array with an initial list of elements";
+  let description = [{
+    This operation creates an array from a list of values. The element on the
+    left-most position in the MLIR assembly format ends up at index 0.
+  }];
+
+  let arguments = (ins Variadic<AnyType>:$elements);
+  let results = (outs ArrayType:$result);
+
+  let hasCustomAssemblyFormat = 1;
+  let hasVerifier = 1;
+}
+
+def ArrayExtractOp : RTGOp<"array_extract", [
```

Comment:
If you want to align on firrtl terminology, the corresponding operation in firrtl is called "subaccess".

Reply:
I aligned it to the HW aggregates, but called it extract like `hw.struct_extract` and `comb.extract` because I think `hw.array_get` is the inconsistent one there.